Firebase/InstanceID/FIRInstanceID.m

```diff
@@ -245,19 +245,21 @@ - (NSString *)cachedTokenIfAvailable {
 }
 
 - (void)setDefaultFCMToken:(NSString *)defaultFCMToken {
-  if (_defaultFCMToken && defaultFCMToken && [defaultFCMToken isEqualToString:_defaultFCMToken]) {
-    return;
+  // Sending this notification out will ensure that FIRMessaging and FIRInstanceID has the updated
+  // default FCM token.
+  // Only notify of token refresh if we have a new valid token that's different than before
+  if ((defaultFCMToken.length && _defaultFCMToken.length &&
+       ![defaultFCMToken isEqualToString:_defaultFCMToken]) ||
+      (defaultFCMToken.length && !_defaultFCMToken.length) ||
```

Comment:
For these two lines, how can there be a length and not a length at the same time?

Reply:
one is parameter that passes in(defaultFCMToken) that is not an empty string and the other is an  instance variable (_defaultFCMToken) that might be empty or nil, in this case, we still think token is changed. Does that make sense? 
Those two parameters are different adn we only assign  the updated value to instance variable after this check.

Reply:
Oh I see. I missed the underscore.

Would `defaultFCMToken.length !=_defaultFCMToken.length` be sufficient to replace lines 253 and 254?

Reply:
Good catch! Updated.